src/Nethermind/Nethermind.Consensus.AuRa/Withdrawals/WithdrawalContractFactory.cs

```diff
@@ -0,0 +1,35 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using Nethermind.Abi;
+using Nethermind.Consensus.AuRa.Contracts;
+using Nethermind.Core;
+using Nethermind.Core.Specs;
+using Nethermind.Evm.TransactionProcessing;
+using Nethermind.Specs.ChainSpecStyle;
+
+namespace Nethermind.Consensus.AuRa.Withdrawals;
+
+public class WithdrawalContractFactory : IWithdrawalContractFactory
+{
+    private readonly IAbiEncoder _abiEncoder;
+    private readonly Address _contractAddress;
+    private readonly ForkActivation _contractTransition;
+
+    public WithdrawalContractFactory(AuRaParameters parameters, IAbiEncoder abiEncoder)
+    {
+        ArgumentNullException.ThrowIfNull(parameters);
+
+        _abiEncoder = abiEncoder ?? throw new ArgumentNullException(nameof(abiEncoder));
+        _contractAddress = parameters.WithdrawalContractAddress;
+        _contractTransition = parameters.WithdrawalContractTransition;
+    }
+
+    public IWithdrawalContract Create(ITransactionProcessor processor)
```

Comment:
@LukaszRozmej @jmederosalvarado The design is similar to `AuRaRewardCalculatorSource.Get` where `ITransactionProcessor` is not injected through the constructor but provided every time. Is that an appropriate way in this case?

Reply:
I think yes, because this will be used from different contexts:
1. main BlockProcessing loop - main TransactionProcessor
2. BlockProduction - needs production TransactionProcessor
3. JsonRpc - ad hoc processors